huntabyte/bits-ui
# feat: Date Field add data-placeholder attribute and delete improv

## Changes

Currently i am porting [Origin UI](https://originui.com/) to Svelte [(See)](https://originui-svelte.pages.dev/). As i was trying to port the Components over i saw that you are handling the placeholder styles with:
```html
<component class="aria-[valuetext=Empty]:something"></component>
```
The [React Aria Datepicker](https://react-spectrum.adobe.com/react-aria/DatePicker.html) changes the `aria-valuetext`based on the locale. So the styling would only work if the locale is`en`. If you decide to implement translation based on the locale in the future, the current implementation would break styling. 

On top of that, as i was writing the tests for that i saw that a `CTRL + A + BACKSPACE / DELETE` deletion was not reliably possible. I don't really know if you want to change that. Again, the React Aria implementation doesn't have that. You can't even `CTRL + A` a Segment (which i find weird). 🤷‍♂️

### 1. TLDR: Improved placeholder handling
- Added `data-placeholder="true"` attribute to empty date/time segments
- Replaced the previous `aria-[valuetext=Empty]:text-muted-foreground` styling approach
- This change makes the placeholder styling more robust and locale-independent
  - Previous implementation relied on `valuetext` which varies based on locale
  - Example: styling would only work if you decide to never implement i18n
  - New approach works consistently, even if changes are happening in the future

### 2. TLDR: Enhanced deletion behavior
- Added proper support for clearing date/time segments when fully selected
- Now supports both:
  - CTRL + A + Backspace
  - CTRL + A + Delete

## Test Coverage

Added two tests:

1. Verification of clearing date segments when fully selected:
 ```typescript
//date-field.test.ts
   it("should clear date segments when fully selected and backspace is pressed", async () => {
     // Tests CTRL + A + Backspace functionality across all segments
   });
```
2. Verification of data-placeholder attribute behavior:
  ```typescript
//date-field.test.ts
it("should apply data-placeholder attribute to empty segments", async () => {
  // Tests initial state, filling, clearing, and partial input scenarios
});
```

### Visual Changes

Empty date/time segments now use data-placeholder attribute for styling
Visual appearance remains the same, but implementation is more robust

### Testing Instructions

Open the `DateField` component (or based on that)
Try selecting all text in any segment (CTRL + A)
Verify deletion works with both:

1. `Backspace` key
2. `Delete` key

Verify placeholder styling appears correctly when segments are empty
